maintenance/wikia/WikiFactoryVariables/migrateWikiFactoryToHttps.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+* Maintenance script to manage WikiFactory variable to https
+* @usage
+* 	# this will migrate wgUploadPath for wiki with ID 119:
+*   /usr/wikia/backend/bin/run_maintenance '--script=wikia/WikiFactoryVariables/migrateWikiFactoryToHttps.php --dry-run --varName wgUploadPath' --id=119
+*/
+
+ini_set( 'display_errors', 'stderr' );
+ini_set( 'error_reporting', E_NOTICE );
+
+require_once( dirname( __FILE__ ) . '/../../Maintenance.php' );
+
+use \Wikia\Logger\WikiaLogger;
+
+/**
+ * Class MigrateWikiFactoryToHttps
+ */
+class MigrateWikiFactoryToHttps extends Maintenance {
+
+	protected $dryRun  = false;
+	protected $verbose = false;
+	protected $varName = '';
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = "Migrates Wiki Factory variable to HTTPS";
+		$this->addOption( 'dry-run', 'Dry run mode', false, false, 'd' );
+		$this->addOption( 'varName', 'Name of the Wiki Factory variable', true, true, 'k' );
+		$this->addOption( 'file', 'File where to save values that are going to be altered', false, true, 'f' );
+	}
+
+	public function execute() {
+		global $wgCityId, $wgMedusaHostPrefix;
+		$this->dryRun  = $this->hasOption( 'dry-run' );
+		$this->varName = $this->getOption( 'varName', '' );
+		$fileName = $this->getOption( 'file', false );
+
+		if ( empty( $this->varName ) ) {
+			$this->error( "Error: Variable name cannot be empty." . PHP_EOL );
+			return false;
+		}
+
+		$varData = WikiFactory::getVarValueByName( $this->varName, $wgCityId, true );
+		if ( !$varData ) {
+			$this->error( "Error: " . $this->varName . " not found." . PHP_EOL );
+			return false;
+		}
+
+		$fh = false;
+		if ( $fileName ) {
+			$fh = fopen( $fileName, "a" );
+			if ( !$fh ) {
+				$this->error( "Could not open file '$fileName' for write!'" . PHP_EOL );
+				return false;
+			}
+		}
+
+		$oldValue = $keyValue = $varData;
+
+		if ( empty( $keyValue ) ) {
+			$this->output( "Variable is empty for $wgCityId - skipping" . PHP_EOL );
+
+			if ( $fh ) {
```

Comment:
This `if`/`fclose` pair is repeated too many times, could you please move it to `finally` section?

Reply:
I did it in a bit different way: 3a984026cf8